auth/auth.go

```diff
@@ -899,29 +899,32 @@ func (auth *Authenticator) authenticateJWTIdentity(identity *Identity, provider
 		return nil, PrincipalConfig{}, time.Time{}, err
 	}
 
+	updates = PrincipalConfig{
+		Name:        base.StringPtr(username),
+		Email:       &identity.Email,
+		JWTIssuer:   &provider.Issuer,
+		JWTRoles:    jwtRoles,
+		JWTChannels: jwtChannels,
+	}
+
+	if user != nil {
+		return user, updates, identity.Expiry, nil
+	}
+
 	// Auto-registration. This will normally be done when token is originally returned
 	// to client by oidc callback, but also needed here to handle clients obtaining their own tokens.
-	if user == nil && provider.Register {
+	if provider.Register {
 		base.DebugfCtx(auth.LogCtx, base.KeyAuth, "Registering new user: %v with email: %v", base.UD(username), base.UD(identity.Email))
-		var err error
-		user, err = auth.RegisterNewUser(username, identity.Email)
+		user, err := auth.RegisterNewUser(username, identity.Email)
 		if err != nil && !base.IsCasMismatch(err) {
 			base.InfofCtx(auth.LogCtx, base.KeyAuth, "Error registering new user: %v", err)
```

Comment:
Is this error going to contain the username? I think in practice this isn't going to error looking at some pathways through `RegisterNewUser`. The best idea for this to fail is if `auth.Save` fails with a write error.

I did notice that if this user has a bad email address we still let the user register. That's outside of this PR, but I wonder if that's the right behavior.

Reply:
Included username in log message.

Are you sure about that email comment though? It's validated in `p.validate()` during save and returns an error.

Reply:
https://github.com/couchbase/sync_gateway/pull/4217#discussion_r318173541